client/src/core/TableClass.js

```diff
@@ -241,6 +251,112 @@ export class Table {
     return run_template(this.title_def[lang]);
   }
 
+  get dimensions() {
+    const cols = _.flatMap(this._cols, (col) =>
+      _.has(col, "children") ? col.children : col
+    );
+    const custom_dimensions = _.chain(cols)
+      .map("custom_groupings")
+      .compact()
+      .head()
+      .keys()
+      .value();
+    return _.concat(
+      "all",
+      _.chain(cols)
+        .filter((col) => col.groupings === undefined)
+        .map("nick")
+        .concat(custom_dimensions)
+        .compact()
+        .value()
+    );
+  }
+
+  is_custom_dim(dimension) {
+    return (
+      !_.chain(this._cols)
+        .flatMap((col) => (_.has(col, "children") ? col.children : col))
+        .map("nick")
+        .includes(dimension)
+        .value() && dimension != "all"
+    );
+  }
+
+  get_dimension_col(dimension) {
+    const cols = _.flatMap(this._cols, (col) =>
+      _.has(col, "children") ? col.children : col
+    );
+
+    return (
+      _.chain(cols)
+        .filter((col) => col.nick === dimension)
+        .head()
+        .value() ||
+      _.chain(cols)
+        .filter((col) => _.has(col, `custom_groupings.${dimension}`))
+        .head()
+        .value() || { nick: "all" }
+    );
+  }
+
+  get_group_by_func() {
+    this.group_by_func = (data, dimension) => {
+      if (!this.is_custom_dim(dimension)) {
+        return _.groupBy(data, dimension);
+      }
+      return _.groupBy(data, (row) =>
+        this.get_dimension_col(dimension).custom_groupings[dimension].group_by(
+          row
+        )
+      );
+    };
+  }
+
+  get_dimension_col_values_func() {
+    this.dimension_col_values_func = (row, dimension) => {
+      const dimension_col = this.get_dimension_col(dimension);
+
+      if (!this.is_custom_dim(dimension)) {
+        return [dimension, row[dimension]];
+      } else {
+        return dimension_col.custom_groupings[dimension].dim_col_value(row);
+      }
+    };
+  }
+
+  // TODO: come up with a shorter, better name for this sum_cols_by_grouped_data
```

Comment:
Seems like a fine name to me. Short only matters if you have to type it _a lot_. Otherwise it's good as long as it's descriptive enough.